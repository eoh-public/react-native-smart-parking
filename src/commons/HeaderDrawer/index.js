import React, { memo } from 'react';
import { View, Image, StyleSheet } from 'react-native';
import FastImage from 'react-native-fast-image';
import { Icon } from '@ant-design/react-native';
import { TESTID } from '../../configs/Constants';

import Text from '../Text';
import { Images, Colors } from '../../configs';
import { useSPSelector } from '../../context';

const HeaderDrawer = memo(() => {
  const user = useSPSelector((state) => state.auth.account.user);
  const userName =
    user && user.name !== null
      ? user.name
      : user && user.email !== null
      ? user.email
      : '';

  const phoneNumber = user ? user.phone_number : '';
  return (
    <View style={[styles.row, styles.logoWrap]}>
      <Image source={Images.logo} style={styles.logo} />

      <View style={styles.avatarBackground}>
        {user.avatar ? (
          <FastImage
            source={{ uri: user.avatar }}
            style={styles.avatar}
            testID={TESTID.FAST_IMAGE_USER_AVATAR}
          />
        ) : (
          <View style={styles.avatar}>
            <Icon name={'user'} size={27} />
          </View>
        )}
        <View style={styles.textContainer}>
          <Text bold type="H4" color={Colors.Gray9} style={styles.fullName}>
            {userName}
          </Text>
          <Text type="Body" color={Colors.Gray8} style={[styles.email]}>
            {phoneNumber}
          </Text>
        </View>
      </View>
    </View>
  );
});

export default HeaderDrawer;

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  logoWrap: {
    paddingTop: 32,
    paddingLeft: 28,
    paddingRight: 23,
  },
  logo: {
    width: 32,
    height: 16,
    position: 'absolute',
    top: 10,
    left: 30,
  },
  avatarBackground: {
    flexDirection: 'row',
    flex: 1,
    paddingVertical: 22,
    borderBottomColor: Colors.Gray4,
    borderBottomWidth: 1,
  },
  avatar: {
    height: 50,
    width: 50,
    borderRadius: 25,
    borderWidth: 0.5,
    borderColor: Colors.Gray5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textContainer: {
    marginLeft: 15,
    justifyContent: 'flex-start',
    flex: 1,
  },
  fullName: {
    marginBottom: 6,
  },
  email: {
    backgroundColor: Colors.TextTransparent,
  },
});
