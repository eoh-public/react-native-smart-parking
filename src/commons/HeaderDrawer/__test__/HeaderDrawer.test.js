import React from 'react';
import { Provider } from 'react-redux';
import renderer, { act } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import HeaderDrawer from '../index';

const mockStore = configureStore([]);

describe('Test HeaderDrawer', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      auth: {
        account: {
          user: {
            name: 'Name',
            email: 'name@gmail.com',
            avatar: 'avatar',
          },
        },
      },
    });

    act(() => {
      component = renderer.create(
        <Provider store={store}>
          <HeaderDrawer />
        </Provider>
      );
    });
  });

  test('create HeaderDrawer', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
