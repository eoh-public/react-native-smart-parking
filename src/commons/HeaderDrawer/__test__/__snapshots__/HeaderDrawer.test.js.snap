// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HeaderDrawer create HeaderDrawer 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
      Object {
        "paddingLeft": 28,
        "paddingRight": 23,
        "paddingTop": 32,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../App/Images/logo.png",
      }
    }
    style={
      Object {
        "height": 16,
        "left": 30,
        "position": "absolute",
        "top": 10,
        "width": 32,
      }
    }
  />
  <View
    style={
      Object {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "paddingVertical": 22,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "alignItems": "center",
            "borderColor": "#D9D9D9",
            "borderRadius": 25,
            "borderWidth": 0.5,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "avatar",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "marginLeft": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 24,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 6,
            },
          ]
        }
      >
        Name
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
